Libraries/LibMedia/Audio/PlaybackStreamWasapi.cpp

```
@@ -0,0 +1,508 @@
+
+    dbgln("Mixing engine audio format:\n");
+    print_audio_format(device_format);
+
+    Vector<Channel, 32> channel_vec;
+    VERIFY(device_format->Format.nChannels <= 32);
+    VERIFY(device_format->dwChannelMask != 0);
+    auto channels = device_format->Format.nChannels;
+
+    // Conveniently the enum is in the same order. The difference is that KSMedia uses a bitmask instaead
+    for (u64 i = 0; i < channels; i++) {
+        u32 mask = 1 << i;
+        if (device_format->dwChannelMask & mask)
+            // i + 1 as the first bit is at i == 0
+            channel_vec.append(static_cast<Channel>(i + 1));
+    }
+    ChannelMap channel_map { channel_vec };
+
+    auto sample_specification = SampleSpecification { device_format->Format.nSamplesPerSec, channel_map };
+    sample_specification_callback(sample_specification);
+
+    // Set up whatever we were given.
+
+    state->wave_format->Format.wFormatTag = WAVE_FORMAT_EXTENSIBLE;
+    state->wave_format->Format.nChannels = device_format->Format.nChannels;
+    state->wave_format->Format.nSamplesPerSec = device_format->Format.nSamplesPerSec;
+    state->wave_format->Format.nAvgBytesPerSec = device_format->Format.nAvgBytesPerSec;
+    state->wave_format->Format.nBlockAlign = device_format->Format.nBlockAlign;
+    state->wave_format->Format.wBitsPerSample = device_format->Format.wBitsPerSample;
+    state->wave_format->Format.cbSize = sizeof(WAVEFORMATEXTENSIBLE) - sizeof(WAVEFORMATEX);
+    state->wave_format->Samples.wValidBitsPerSample = device_format->Samples.wValidBitsPerSample;
+    state->wave_format->dwChannelMask = device_format->dwChannelMask;
+    state->wave_format->SubFormat = device_format->SubFormat;
+
+    // FIXME: From what I understand the audio mixer uses 32bit floats internally. If that is not always the case we need to handle it.
+    VERIFY(device_format->SubFormat == KSDATAFORMAT_SUBTYPE_IEEE_FLOAT);
+    VERIFY(device_format->Format.wBitsPerSample == 32);
```

Instead of verifying this, could we just create a format that reuses the sample rate and channel count/mask, but with the float subformat? Then we don't need a verify here, and we can just return an error if the API rejects the format.